packages/hooks/src/useRequestPhoneNumberOTP.tsx

```diff
@@ -0,0 +1,58 @@
+import React from 'react';
+import { useMutation } from '@deriv/api';
+import { VERIFICATION_SERVICES } from '@deriv/shared';
+import { Localize } from '@deriv/translations';
+
+type THandleError = {
+    code: string;
+    message: string;
+};
+
+/** A hook for requesting OTP which is sent on whatsapp or sms platforms */
+const useRequestPhoneNumberOTP = () => {
+    const { data, mutate, ...rest } = useMutation('phone_number_challenge');
+    const [error_message, setErrorMessage] = React.useState<React.ReactNode>('');
+
+    const requestOnSMS = () => {
+        mutate({ payload: { carrier: VERIFICATION_SERVICES.SMS } });
+    };
+
+    const requestOnWhatsApp = () => {
+        mutate({ payload: { carrier: VERIFICATION_SERVICES.WHATSAPP } });
+    };
+
+    const handleError = ({ code, message }: THandleError) => {
```

Comment:
This is more like a formatError function.isnt?

Reply:
yeap this is a format error function

Reply:
can we rename function to formatError then?